packages/openapi-v3/src/controller-spec.ts

```diff
@@ -148,34 +161,38 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
     spec.paths[path][verb] = operationSpec;
 
     debug(`  inferring schema object for method %s`, op);
-    const paramTypes = MetadataInspector.getDesignTypeForMethod(
+    const opMetadata = MetadataInspector.getDesignTypeForMethod(
       constructor.prototype,
       op,
-    ).parameterTypes;
+    );
+    const paramTypes = opMetadata.parameterTypes;
 
     const isComplexType = (ctor: Function) =>
-      !_.includes([String, Number, Boolean, Array, Object], ctor) &&
-      !isReadableStream(ctor);
+      !_.includes([String, Number, Boolean, Array, Object], ctor);
 
     for (const p of paramTypes) {
       if (isComplexType(p)) {
-        if (!spec.definitions) {
-          spec.definitions = {};
+        if (!spec.components) {
+          spec.components = {};
+        }
+        if (!spec.components.schemas) {
+          spec.components.schemas = {};
         }
         const jsonSchema = getJsonSchema(p);
         const openapiSchema = jsonToSchemaObject(jsonSchema);
-
         if (openapiSchema.definitions) {
           for (const key in openapiSchema.definitions) {
-            spec.definitions[key] = openapiSchema.definitions[key];
+            spec.components.schemas[key] = openapiSchema.definitions[key];
           }
           delete openapiSchema.definitions;
         }
 
-        spec.definitions[p.name] = openapiSchema;
+        spec.components.schemas[p.name] = openapiSchema;
         break;
       }
     }
+    // const returnType = opMetadata.returnType;
+    // buildResponsesSpec(returnType, spec);
```

Comment:
We usually don't keep code that's commented out. Can you remove it please?

Reply:
oops, was trying to add response spec then realized it's out of the scope. ^ didn't clean all related code. deleted.